AppDevPods/AppDevCameraKit/ADKCamera.m

```diff
@@ -0,0 +1,1342 @@
+        } else {
+            self.videoCaptureDevice = [self captureDeviceWithCameraPosition:captureDevicePosition];
+        }
+
+        // Initialize AVCaptureDeviceInput
+        NSError *error = nil;
+        self.videoCaptureDeviceInput = [AVCaptureDeviceInput deviceInputWithDevice:self.videoCaptureDevice error:&error];
+        if (error) {
+            [self handleErrorWithCode:error.code userInfo:error.userInfo];
+            return;
+        }
+
+        if ([self.captureSession canAddInput:self.videoCaptureDeviceInput]) {
+            [self.captureSession addInput:self.videoCaptureDeviceInput];
+        } else {
+            [self handleErrorWithCode:ADKCameraErrorCodeSession description:@"The session can't add capture input in initialized flow."];
+            return;
+        }
+
+        self.captureStillImageOutput = [[AVCaptureStillImageOutput alloc] init];
+        NSDictionary *outputSettings = @{AVVideoCodecKey: AVVideoCodecJPEG};
+        self.captureStillImageOutput.outputSettings = outputSettings;
+        if ([self.captureSession canAddOutput:self.captureStillImageOutput]) {
+            [self.captureSession addOutput:self.captureStillImageOutput];
+        } else {
+            [self handleErrorWithCode:ADKCameraErrorCodeSession description:@"The session can't add capture output in initialized flow."];
+            return;
+        }
+
+        if (self.camcoderMode) {
+            self.audioCaptureDevice = [AVCaptureDevice defaultDeviceWithMediaType:AVMediaTypeAudio];
+            self.audioCaptureDeviceInput = [AVCaptureDeviceInput deviceInputWithDevice:self.audioCaptureDevice error:&error];
+
+            if (error) {
+                [self handleErrorWithCode:error.code userInfo:error.userInfo];
+            }
+
+            if ([self.captureSession canAddInput:self.audioCaptureDeviceInput]) {
+                [self.captureSession addInput:self.audioCaptureDeviceInput];
+            }
+
+            self.captureMovieFileOutput = [[AVCaptureMovieFileOutput alloc] init];
+
+            if ([self.captureSession canAddOutput:self.captureMovieFileOutput]) {
+                [self.captureSession addOutput:self.captureMovieFileOutput];
+            }
+        }
+    }
+}
+
+
+#pragma mark -- Help methods
+
+- (BOOL)flashAvailable
+{
+    if (!self.videoCaptureDevice) {
+        return NO;
+    }
+
+    return self.videoCaptureDevice.hasFlash && self.videoCaptureDevice.isFlashAvailable;
+}
+
+- (BOOL)torchAvailable
+{
+    if (!self.videoCaptureDevice) {
+        return NO;
+    }
+
+    return self.videoCaptureDevice.hasTorch && self.videoCaptureDevice.isTorchAvailable;
+}
+
+
+#pragma mark -- Camera feature methods
+
+- (void)optimizeForHighestFrameRate
+{
+    if (self.videoCaptureDevice) {
+        [self configureCameraForHighestFrameRate:self.videoCaptureDevice];
+    }
+}
+
+- (void)startCamera
+{
+    if (self.captureSession) {
+        [self.captureSession startRunning];
+        [self updateCameraStatus];
```

Comment:
I have implemented a delegate that called [ADKCamera:sessionIsRunnig:] to notify developer the session is running. The developer can adopt this delegate to trigger his own behavior.